win32ss/user/ntuser/clipboard.c

```diff
@@ -1208,18 +1212,19 @@ NtUserConvertMemHandle(
     }
     _SEH2_EXCEPT(EXCEPTION_EXECUTE_HANDLER)
     {
-        pMemObj = NULL;
+        Status = _SEH2_GetExceptionCode();
     }
     _SEH2_END;
 
     /* Release the extra reference (UserCreateObject added 2 references) */
     UserDereferenceObject(pMemObj);
 
     /* If we failed to copy data, remove handle */
-    if (!pMemObj)
+    if (!NT_SUCCESS(Status))
     {
         UserDeleteObject(hMem, TYPE_CLIPDATA);
         hMem = NULL;
+        SetLastNtError(Status);
```

Comment:
Are you sure this NtUserConvertMemHandle function modifies the last error? Please add a test for this.